From f2fa75accbd62a656f78b8467cfdb59e0510b0e9 Mon Sep 17 00:00:00 2001
From: Seungha Yang <seungha@centricular.com>
Date: Thu, 2 Sep 2021 22:06:52 +0900
Subject: [PATCH] videoparseutils: Fix for wrong CEA708 minimum size check

The minimum possible size of valid CEA708 data is 3 bytes, not 7 bytes

Part-of: <https://gitlab.freedesktop.org/gstreamer/gst-plugins-bad/-/merge_requests/2505>
---
 gst/videoparsers/gstvideoparseutils.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/gst/videoparsers/gstvideoparseutils.c b/gst/videoparsers/gstvideoparseutils.c
index cec724ac29..4e94d21b55 100644
--- a/gst/videoparsers/gstvideoparseutils.c
+++ b/gst/videoparsers/gstvideoparseutils.c
@@ -142,7 +142,11 @@ gst_video_parse_user_data (GstElement * elt, GstVideoParseUserData * user_data,
       }
       switch (user_data_type_code) {
         case A53_USER_DATA_TYPE_CODE_CC_DATA:
-          if (gst_byte_reader_get_remaining (br) < 7) {
+          /* 1 (cc count byte) +
+           * 1 (reserved byte, 0xff) +
+           * 1 (marker_bits, 0xff)
+           */
+          if (gst_byte_reader_get_remaining (br) < 3) {
             GST_WARNING_OBJECT (elt,
                 "Closed caption data packet too short, ignoring");
             break;
-- 
GitLab

